src/generate.ml4

```diff
@@ -322,26 +317,27 @@ let lemma_kind = (Decl_kinds.Global, false, Decl_kinds.Proof Decl_kinds.Lemma)
 
 let dummy_hook = Lemmas.mk_hook (fun _ _ -> ())
 
-let get_context ty = 
+let get_context ty =
   let env = Global.env () in
   let sigma = Evd.from_env env in
-  let sigma, _ty = Typing.type_of env sigma ty in
+  let sigma, _ty = Typing.type_of env sigma (EConstr.of_constr ty) in
    sigma
 
 let prove_refl indconstr mind body =
   let id_t = body.Declarations.mind_typename in
   let id_eq = add_suffix id_t "_eq" in
   let x = Nameops.make_ident "x" None in
-  let ceq = Universes.constr_of_reference
+  let ceq = Universes.constr_of_global
```

Comment:
I am not sure whether `Universes.constr_of_global` is OK here, because it is used on a proof of reflexivity. Would this be the right way to do it?
```
let env = Global.env () in
let evd = Evd.from_env env in
let (evd',ceq) = Evarutil.new_global evd (Nametab.global (Libnames.Ident (dl id_eq))) in
let eceq = EConstr.to_constr evd' ceq

```